packages/sanity/src/presentation/loader/LiveQueries.tsx

```diff
@@ -195,69 +114,35 @@ export default function LoaderQueries(props: LoaderQueriesProps): React.JSX.Elem
   )
   useEffect(() => {
     if (comlink) {
-      // eslint-disable-next-line @typescript-eslint/no-shadow
-      const {projectId, dataset} = clientConfig
       comlink.post('loader/perspective', {
-        projectId: projectId!,
-        dataset: dataset!,
+        projectId,
+        dataset,
         perspective: activePerspective,
       })
     }
-  }, [comlink, clientConfig, activePerspective])
-
-  const handleSyncTags = useEffectEvent((event: LiveEventMessage) => {
-    const flattenedSyncTags = Array.from(syncTagsInUse).flat()
-    const hasMatchingTags = event.tags.some((tag) => flattenedSyncTags.includes(tag))
-    if (hasMatchingTags) {
-      setLastLiveEventId(event.id)
-    } else {
-      // eslint-disable-next-line no-console
-      console.log('No matching tags found', event.tags, {flattenedSyncTags})
-    }
-  })
-  useEffect(() => {
-    const liveClient = createClient(client.config()).withConfig({
-      // Necessary for the live drafts to work
-      apiVersion: 'vX',
-    })
-    const subscription = liveClient.live
-      .events({includeDrafts: true, tag: 'presentation-loader'})
-      .subscribe({
-        next: (event) => {
-          if (event.type === 'message') {
-            handleSyncTags(event)
-          } else if (event.type === 'restart') {
-            setLastLiveEventId(event.id)
-          } else if (event.type === 'reconnect') {
-            setLastLiveEventId(null)
-          }
-        },
-        // eslint-disable-next-line no-console
-        error: (err) => console.error('Error validating EventSource URL:', err),
-      })
-    return () => subscription.unsubscribe()
-  }, [client, handleSyncTags])
+  }, [comlink, activePerspective, projectId, dataset])
 
   /**
    * Defer the liveDocument to avoid unnecessary rerenders on rapid edits
    */
-  const liveDocument = useDeferredValue(_liveDocument)
+  const liveDocument = useDeferredValue(props.liveDocument)
+
+  const liveEvents = useLiveEvents(client)
 
   return (
     <>
-      {Object.entries(liveQueries).map(([key, {query, params, perspective}]) => (
+      {[...liveQueries.entries()].map(([key, {query, params, perspective}]) => (
         <QuerySubscription
-          key={`${key}${perspective}`}
-          projectId={clientConfig.projectId!}
-          dataset={clientConfig.dataset!}
+          key={`${liveEvents.resets}:${key}`}
```

Comment:
The `liveEvents.resets` ensures that if a `reconnect` or `restart` live event happens then `QuerySubscription` components here start from a clean slate. This helps ensure their internal tracking of what `liveEventsMessages` they can ignore, as they happened before the component mounted, and which ones to look for a `lastLiveEventId`, works correctly.